tests/js/utils.js

```diff
@@ -204,13 +210,23 @@ export const provideModules = ( registry, extraData = [] ) => {
 		return { ...acc, [ module.slug ]: module };
 	}, {} );
 
+	const moduleIconMap = {
+		adsense: AdsenseIcon,
+		analytics: AnalyticsIcon,
+		optimize: OptimizeIcon,
+		'pagespeed-insights': PagespeedInsightsIcon,
+		'search-console': SearchConsoleIcon,
+		tagmanager: TagManagerIcon,
+	};
+
 	const moduleSlugs = coreModulesFixture.map( ( { slug } ) => slug );
 	const modules = coreModulesFixture
 		.map( ( module ) => {
+			const icon = moduleIconMap[ module.slug ];
 			if ( extraModules[ module.slug ] ) {
-				return { ...module, ...extraModules[ module.slug ] };
+				return { ...module, ...extraModules[ module.slug ], icon };
 			}
-			return { ...module };
+			return { ...module, icon };
```

Comment:
While this technically works, it doesn't really fit here, since `receiveGetModules` has the purpose of receiving module data from the API, i.e. PHP. JS-specific data (like `icon` is) should be registered via `registerModule`. They'll also be stored separately internally in the datastore (`serverDefinitions` vs `clientDefinitions`), and we should stick to that here.

I suggest we introduce a `provideModuleRegistrations` or similar test utility function which calls `registerModule` for every one of the 6 modules by default, with the expected data, allowing for custom overrides to be optionally passed (basically similarly to how this `provideModules` works.

Tests or Storybook stories that require JS module data to be registered should then use that utility function.

Reply:
@felixarntz I have implemented the suggested test utility.